### DISCUSSION TITLE
Can NextJS not handle Page data fetching server side that considers query params?

### DISCUSSION THREAD

**Discussion by AthenaProductions-Adam** [vercel/next.js#64845] [Help]
### Summary

I've search the docs and reviewed everything relevant that i could find but this seems to be something really obvious that's apparently not possible in NextJS, can anyone either confirm if this is the case and maybe suggest a good workaround or tell me what i've missed please?

I'm using the app router and I have a server side page that fetches initial page data, returns this to the client and then there's filters on the page to filter the data in a grid. I need someone to be able to store a url including query params of their current filter, then be able to revisit the page using that url again and everything is filtered how it was before.

In my head the logical flow would be:
- Use SSR to load the initial page with all data retrieved server side, as this is likely cached it's very quick then they don't need for the request waterfall client side to get the data and load it in.
- Any further filtering uses fetch requests to hit a route handler to get retrieve new data.

I've noticed two issues in the docs that seem obvious to me but i can't really find anyone else having the problem, so i have a couple of questions:
1. There's no examples of handling fetch requests client side, have i missed this in the documentation somewhere? Viewing the docs regarding fetching data client side refers you to the route handler section, but this doesn't include a single example of the fetch request being made by a client component, it only shows server side code. This seems like something that would make the example scenarios more complete in my mind.
2. As far as i can tell from the docs, page requests in SSR have no way to check the URL used, is that something NextJS just doesn't allow? Coming from a background of making a lot of server side apps in other frameworks, this seems crazy to me. I can't think of a reason why this would be an issue, the request hits the server and the NextJS obviously uses the URL to decide on the correct route, so i'm not sure why being able to read the query params would be a bad thing? It means i can't use SSR fully as my initial render cannot load the data required as it needs to consider the URL query params in order to load the data correctly.

### Additional information

```js
NextJS version: 14.1.0
Node Version: 20.8.0
```

**Comment by GabenGar** [comment]
Can you provide a basic example of what you are trying to achieve?

**Reply by AthenaProductions-Adam** [reply]
While starting to get you an example, i think i've just found what i needed, i think i need searchParams in my server component page, not sure how i missed this previously.

https://nextjs.org/docs/app/api-reference/file-conventions/page#searchparams-optional

Sorry to waste time. I think because I searched in the docs for `query`, it never brought up this file conventions page as it never refers to them as `query params`, only as `search params`, I'm used to the term `query` rather than `search`.

**Reply by GabenGar** [reply]
They are search params because the interface responsible for them is called [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams).
`query` is a bit too ambiguous, because variable parts in the pathname are also called "queries", but also old NodeJS interface responsible for parsing them (quite liberally) was called [`querystring`](https://nodejs.org/api/querystring.html).
